bigquery/google/cloud/bigquery/magics.py

```diff
@@ -265,6 +276,14 @@ def _cell_magic(line, query):
     """
     args = magic_arguments.parse_argstring(_cell_magic, line)
 
+    if args.params is not None:
+        try:
+            params = json.loads(args.params)
+        except JSONDecodeError as e:
+            raise JSONDecodeError('--params is not a correctly formatted JSON string', e.doc, e.pos)
```

Comment:
Can you clarify the utility of replacing the original exception here?

Reply:
The reason for replacing is that the original exception would be something like this:

```
<stack trace>
JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 28 (char 27)
```

In order to know where the error is, you have to follow the stacktrace. With this message, it makes it clearer where the error was.